braid/uvector.c

```diff
@@ -201,6 +201,45 @@ _braid_UGetVector(braid_Core         core,
    return _braid_error_flag;
 }
 
+/* Retrieve the last time-step vector.  Note that the
+ * last time-step is only available after Braid is 
+ * finished cycling and done is True.  This routine should
+ * only be called by the last processor in time. */
+braid_Int
+_braid_UGetLast(braid_Core        core,
+                braid_BaseVector *u_ptr)
+{
+   _braid_Grid       **grids    = _braid_CoreElt(core, grids);
+   MPI_Comm            comm     = _braid_CoreElt(core, comm);
+   braid_Int           cfactor  = _braid_GridElt(grids[0], cfactor);
+   braid_Int           gupper   = _braid_CoreElt(core, gupper);
+   braid_Int           done     = _braid_CoreElt(core, done);
+   braid_BaseVector    ulast;
+   braid_Int           num_procs, myid;
+
+   MPI_Comm_size( comm, &num_procs);
+   MPI_Comm_rank( comm, &myid );
+   
+   /* If Done, get vector at last time step*/
+   if(done && (myid == (num_procs-1) ) )
```

Comment:
@rfalgout @steffi7574 
As you can see, I'm only making ulast available if "done" is true.  USetVector only stores into ulast if done is true.  If this will conflict with any desired functionality, let me know.